Objects/frameobject.c

```diff
@@ -148,8 +148,9 @@ framelocalsproxy_setitem(PyObject *self, PyObject *key, PyObject *value)
     if (PyUnicode_CheckExact(key)) {
         int i = framelocalsproxy_getkeyindex(frame, key, false);
         if (i >= 0) {
-            _PyLocals_Kind kind = _PyLocals_GetKind(co->co_localspluskinds, i);
+            _Py_Executors_InvalidateDependency(PyInterpreterState_Get(), co, 1);
```

Comment:
The way I solved the `_Py_TIER2` dependency elsewhere is to put this call itself inside `#ifdef _Py_TIER2`. Then you don't need the dummy version you added to optimizer.c.